schema/testdata.go

```diff
@@ -141,119 +252,33 @@ type GenTestDataOptions struct {
-				}
-			} else if arrow.TypeEqual(c.Type, arrow.ListOf(arrow.PrimitiveTypes.Int64)) {
-				bldr.Field(i).(*array.ListBuilder).Append(true)
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*array.Int64Builder).Append(1)
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*array.Int64Builder).Append(2)
-			} else if arrow.TypeEqual(c.Type, arrow.FixedWidthTypes.Timestamp_us) {
-				if c.Name == CqSyncTimeColumn.Name {
-					bldr.Field(i).(*array.TimestampBuilder).Append(arrow.Timestamp(opts.SyncTime.UTC().Truncate(time.Millisecond).UnixMicro()))
-				} else {
-					t := time.Now()
-					if !opts.StableTime.IsZero() {
-						t = opts.StableTime
-					}
-					bldr.Field(i).(*array.TimestampBuilder).Append(arrow.Timestamp(t.UTC().Truncate(time.Millisecond).UnixMicro()))
-				}
-			} else if arrow.TypeEqual(c.Type, types.ExtensionTypes.JSON) {
-				bldr.Field(i).(*types.JSONBuilder).Append(map[string]any{"test": "test"})
-			} else if arrow.TypeEqual(c.Type, arrow.ListOf(types.ExtensionTypes.UUID)) {
-				bldr.Field(i).(*array.ListBuilder).Append(true)
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*types.UUIDBuilder).Append(uuid.MustParse("00000000-0000-0000-0000-000000000001"))
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*types.UUIDBuilder).Append(uuid.MustParse("00000000-0000-0000-0000-000000000002"))
-			} else if arrow.TypeEqual(c.Type, types.ExtensionTypes.Inet) {
-				_, ipnet, err := net.ParseCIDR("192.0.2.0/24")
-				if err != nil {
-					panic(err)
-				}
-				bldr.Field(i).(*types.InetBuilder).Append(ipnet)
-			} else if arrow.TypeEqual(c.Type, arrow.ListOf(types.ExtensionTypes.Inet)) {
-				bldr.Field(i).(*array.ListBuilder).Append(true)
-				_, ipnet, err := net.ParseCIDR("192.0.2.1/24")
-				if err != nil {
-					panic(err)
-				}
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*types.InetBuilder).Append(ipnet)
-				_, ipnet, err = net.ParseCIDR("192.0.2.1/24")
-				if err != nil {
-					panic(err)
-				}
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*types.InetBuilder).Append(ipnet)
-			} else if arrow.TypeEqual(c.Type, types.ExtensionTypes.Mac) {
-				mac, err := net.ParseMAC("aa:bb:cc:dd:ee:ff")
-				if err != nil {
-					panic(err)
-				}
-				bldr.Field(i).(*types.MacBuilder).Append(mac)
-			} else if arrow.TypeEqual(c.Type, arrow.ListOf(types.ExtensionTypes.Mac)) {
-				mac, err := net.ParseMAC("aa:bb:cc:dd:ee:ff")
-				if err != nil {
-					panic(err)
-				}
-				bldr.Field(i).(*array.ListBuilder).Append(true)
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*types.MacBuilder).Append(mac)
-				mac, err = net.ParseMAC("11:22:33:44:55:66")
-				if err != nil {
-					panic(err)
-				}
-				bldr.Field(i).(*array.ListBuilder).ValueBuilder().(*types.MacBuilder).Append(mac)
-			} else {
-				panic("unknown type: " + c.Type.String() + " column: " + c.Name)
+			example := getExampleJSON(c.Name, c.Type, opts)
```

Comment:
I think you mentioned it but I can't recall, why you went with `json` instead of regular `append` to builder?

Reply:
Both can work, and I don't recall the exact reason anymore either 😅  I think it was just easier and cleaner to split it into a separate recursive function that can also handle composite types, and doesn't require any type switches. Using JSON is also a common pattern in the Arrow library testing files, which is where I got the idea.